code/implementation/docker/compute/src/main/java/io/cattle/platform/docker/process/instance/DockerInstancePreCreate.java

```diff
@@ -59,11 +56,16 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
         data.put(DockerInstanceConstants.FIELD_NETWORK_MODE, mode);
 
         Network network = networkService.resolveNetwork(instance.getAccountId(), mode);
+        // I don't have too much confidence about deleting this block of code, but there is a scenario which mode isn't belong to any of 
+        // [bridge,host,container,managed]. This happens when we create containers using docker-compose, network mode return by docker 
+        // inspect is actually something like [projectDir-default], which equals to docker bridge
+        /*
         if (network == null && StringUtils.isNotBlank(mode)) {
```

Comment:
Just add a check to only do this for `instance.isNativeContainer`.  So if it's backported we just leave it as is, it it's deployed from rancher we delete.  We need this check to make sure a container doesn't deploy for the ipsec/vxlan network before the network has been created.